src/_guides/language/numbers.md

```diff
@@ -0,0 +1,547 @@
+---
+
+A Flutter app written for iOS or Android
+should just work on the web,
+as long as the app doesn't rely on platform-specific libraries.
+Sometimes, though,
+you might see differences between platforms because
+the Dart implementation of numbers is slightly different on the web.
+
+Although you don't usually need to think about
+these number implementation differences,
+it's good to know that they exist.
+This page has details about the differences
+and how to deal with them.
+
+{{ site.alert.secondary }}
+  **Number implementations in Dart and other languages**
+
+  Dart has always allowed platform-specific representations
+  and semantics for numbers, for reasons of 
+  performance, code size, and platform interoperability.
+
+  Similarly, in C/C++ the commonly used `int` type for integer values is
+  platform-specific to best map to the native machine architecture
+  (16-, 32-, or 64-bit).
+  In Java, the `float` and `double` types for fractional values
+  were originally designed to strictly follow IEEE 754 on all platforms,
+  but this constraint was loosened almost immediately for efficiency reasons
+  (`strictfp` is required for exact coherence).
+{{ site.alert.end }}
+
+
+## Dart number representation
+
+In Dart, all numbers are part of the common `Object` type hierarchy,
+and there are two concrete, user-visible numeric types:
+`int`, representing integer values, and `double`, representing fractional values.
+
+```
+  Object
+    |
+   num
+  /    \
+int   double
+```
+***[TODO: Make this a real figure]***
+
+Depending on the platform,
+those numeric types have different, hidden implementations.
+In particular, Dart has two very different types of targets it compiles to:
+
+* **Native:** Most often, a 64-bit mobile or desktop processor.
+* **Web:** JavaScript as the primary execution engine.
+
+The following table shows how Dart numbers are usually implemented:
+
+<table class="table table-striped nowrap">
+  <tr>
+   <th>Representation</th>
+   <th>Native <code>int</code></th>
+   <th>Native <code>double</code></th>
+   <th>Web <code>int</code></th>
+   <th>Web <code>double</code></th>
+  </tr>
+  <tr>
+   <td><a href="https://en.wikipedia.org/wiki/Two%27s_complement">
+     64-bit signed two’s complement</a>
+   </td>
+   <td>✅</td>
+   <td></td>
+   <td></td>
+   <td></td>
+  </tr>
+  <tr>
+   <td>
+     <a href="https://en.wikipedia.org/wiki/Double-precision_floating-point_format">64-bit floating point</a>
+   </td>
+   <td></td>
+   <td>✅</td>
+   <td>✅</td>
+   <td>✅</td>
+  </tr>
+</table>
+
+***[QUESTION: Could we simplify this section? The \* classes are confusing.
+Maybe we could have a single figure for the platform-specific types
+that shows the native & web class hierarchies side by side?
+Then we could just point out, hey look, int & double can be the same type
+in a web program. It might be easier for me to explain if I could see
+examples of the underlying types you're talking about.]***
```

Comment:
Here's an example of native versions of `int`:

https://github.com/dart-lang/sdk/blob/master/sdk/lib/_internal/vm/lib/integers.dart#L7

Note, there are a few classes (representations) here that "implement" `int`.  I didn't want to get derailed into discussing them here as they're orthogonal to the point of the article - they're more about optimizations than semantics.

Here's similar for web:

https://github.com/dart-lang/sdk/blob/master/sdk/lib/_internal/js_runtime/lib/js_number.dart#L489